packages/sentry/src/adapters/sentryMonitoringAdapter.ts

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import {
+  LogContext,
+  MonitoringAdapter,
+  Severity,
+  Transaction,
+  TransactionContext,
+} from '@monkvision/monitoring';
+
+import * as Sentry from '@sentry/react';
+import { Span } from '@sentry/types';
+
+/**
+ * The status of an Transaction/Span.
+ */
+export enum MonitoringStatus {
+  /** The operation completed successfully. */
+  OK = 'ok',
+  /** Unknown. Any non-standard HTTP status code. */
+  UNKNOWN_ERROR = 'unknown_error',
+  /** The operation was cancelled (typically by the user). */
+  CANCELLED = 'cancelled',
+  /** The operation was aborted, typically due to a concurrency issue. */
+  ABORTED = 'aborted',
+}
+
+/**
+ * Configs required when instantiating the Sentry Monitoring Adapter.
+ */
+export interface SentryMonitoringConfigs {
+  /**
+   * DSN key for sentry.io application
+   *
+   * @default ""
+   */
+  dsn: string;
+  /**
+   * The environment of your application (e.g. "production")
+   *
+   * @default ""
+   */
+  environment: string;
+  /**
+   * Enable debug functionality in the SDK itself
+   *
+   * @default false
+   */
+  debug: boolean;
+  /**
+   * Release version of application
+   *
+   * @default ""
+   */
+  release: string;
+  /**
+   * Sample rate to determine trace sampling.
+   *
+   * 0.0 = 0% chance of a given trace being sent (send no traces) 1.0 = 100% chance of a given trace being sent (send
+   * all traces)
+   *
+   * Tracing is enabled if either this or `tracesSampler` is defined. If both are defined, `tracesSampleRate` is
+   * ignored.
+   *
+   * @default 0.025
+   */
+  tracesSampleRate: number;
+  /**
+   * Custom tags to add in all transaction.
+   *
+   */
+  customTags?: { [tag: string]: string };
+}
+
+const defaultOptions: SentryMonitoringConfigs = {
+  dsn: '',
+  environment: 'local',
+  debug: false,
+  tracesSampleRate: 0.025,
+  release: '',
+};
+
+/**
+ * This is a Sentry Monitoring Adapter that logs element in the Sentry.
+ * There are four methods implemented which are `setUserId`, `log`, `handleError` and `createTransaction`,
+ *
+ * At the initialization level, user have to pass required sentry configuration keys to make connection between
+ * application and Sentry. `log` and `handleError` methods will log data and errors respectively in the Sentry dashboards.
+ * While `createTransaction` method used to measure a performance of an application at any given point.
+ */
+export class SentryMonitoringAdapter implements MonitoringAdapter {
+  constructor(optionsParam: Partial<SentryMonitoringConfigs>) {
```

Comment:
Instead of `Partial<SentryMonitoringConfigs>`, what I suggest you do is create an interface like that :

```typescript
/**
 * Type definition for the config options given to the SentryMonitoringAdapter constructor.
 */
export interface SentryMonitoringAdapterConfig extends Partial<SentryMonitoringConfig> {
  /*
   * DSN key for sentry.io application.
   */
  dsn: string;
}
```

and then use it in the constructor : 

```typescript
constructor(optionsParam: SentryMonitoringAdapterConfig) {
  ...
}
```